### PR TITLE
upgrade rand

### PR DESCRIPTION
**Pull request by dirvine**
Upgrade to latest rand, only affects tests

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @Fraser999

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: Cargo.toml [hunk 1]
```diff
@@ -14,4 +14,4 @@ edition = "2018"
 fake_clock = { version = "~0.3.0", optional = true }
 
 [dev-dependencies]
-rand = "~0.4.1"
+rand = "0.6.5"
```
Comment (ustulation) [inline Cargo.toml]: `~` missing in semver

File: src/lib.rs [hunk 2]
```diff
@@ -336,7 +333,7 @@ where
     }
 
     /// Gets the given key's corresponding entry in the map for in-place manipulation.
-    pub fn entry(&mut self, key: Key) -> Entry<Key, Value> {
+    pub fn entry(&mut self, key: Key) -> Entry<'_, Key, Value> {
```
Comment (ustulation) [inline src/lib.rs]: Not exactly a review, but just for curiosity, is `'_` necessary ? I have not used these but only read about them and the idea seems to be to simplify verbosity by not forcing to mention lifetimes in function signatures before using them. All unmentioned lifetimes are assumed to be bound to the function signature. So is it OK to not mention these or does that lead to compilation OR linter errors ?

Comment (dirvine) [reply src/lib.rs]: No compiler errors, it is a lint issue only with cargo fix using edition-idioms. So no biggies, they can be inferred by the compiler and are, but for readers of the code it is supposed to make the lifetimes more apparent AFAIK.

**Comment by ustulation** [comment]
Reviewed and approved, but needs rebased to allow merging

**Pull request merged**